packages/@ember/-internals/routing/lib/services/router.ts

```diff
@@ -61,19 +65,30 @@ function cleanURL(url: string, rootURL: string) {
    @class RouterService
  */
 export default class RouterService extends Service {
-  _router!: EmberRouter;
+  get _router(): EmberRouter {
+    let router = this[ROUTER];
+    if (router !== undefined) {
+      return router;
+    }
+    const owner = getOwner(this) as any;
+    router = owner.lookup('router:main');
```

Comment:
Can we cast these to the correct values?

```suggestion
    const owner = getOwner(this) as Owner;
    router = owner.lookup('router:main') as EmberRouter;
```

Not sure if I was missing something here...